src/server/ttl/ttl_compaction_filter.rs

```diff
@@ -83,7 +97,11 @@ impl<F: KvFormat> CompactionFilter for TtlCompactionFilter<F> {
             Ok(RawValue {
                 expire_ts: Some(expire_ts),
                 ..
-            }) if expire_ts <= self.ts => CompactionFilterDecision::Remove,
+            }) if expire_ts <= self.ts => {
+                TTL_EXPIRE_KV_SIZE_COUNTER.inc_by(key.len() as u64 + value.len() as u64);
```

Comment:
Suggest to accumulate the metrics in member variables first, then add to the counters on dropping `TtlCompactionFilter`. That would slightly improve performance as prometheus counters are atomic variables unerlying.